### ISSUE TITLE
file-system-route-api not passing params value correctly to config function

### ISSUE THREAD

**Issue by jmolivas** [gatsbyjs/gatsby#34949]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I am working with the file-system-route-api and adding the async config function to defer some of the pages.

I created my pages like this:
```
src/pages
├── {nodeArticle.path__alias}.js
└── {nodePage.path__alias}.js
```

Then my {nodeArticle.path__alias}.js file looks like:

```
export async function config() {
  const { data } = graphql`  
  {
    dsgArticles: allNodeArticle (filter:{field_rendering_mode:{eq:"DSG"}}) {
      nodes {
        path {
          alias
        }
      }
    }
  }
  `

  return ({ params }) => {
    console.log(params);
    return {
      defer: data.dsgArticles.nodes.some( article => article.path.alias === `/${params.path__alias}` )
    }
  }
}
```

### Reproduction Link

https://github.com/octahedroid/gatsbyconf

### Steps to Reproduce

1. Clone repo
2. Execute `yarn build`
3. See output of params 
...

### Expected Result

I can confirm this is a bug since I did rename one of those page alias removing the `/` ie `/blog/old-post-we-want-use-dsg` to `blog-old-post-we-want-use-dsg` the params values show correctly and I was able to implement DSG.

![image](https://user-images.githubusercontent.com/366275/156048680-9ac00660-7cda-4b99-8e25-6172b008c3b6.png)

### Actual Result

The value I am getting form params it's always `blog` instead of `blog/my-page-alias`. 

The `params` value it's removing any route value after the `/`
![image](https://user-images.githubusercontent.com/366275/156048615-673d356c-08a0-4e13-aaab-68ac42d1fe09.png)

### Environment

```shell
System:
    OS: macOS 12.2
    CPU: (10) arm64 Apple M1 Max
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.18.2 - /opt/homebrew/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 6.14.15 - /opt/homebrew/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 98.0.4758.109
    Firefox: 96.0.2
    Safari: 15.3
  npmPackages:
    gatsby: ^4.8.1 => 4.8.1 
    gatsby-plugin-image: ^2.8.1 => 2.8.1 
    gatsby-plugin-postcss: ^5.8.0 => 5.8.0 
    gatsby-plugin-react-helmet: ^5.8.0 => 5.8.0 
    gatsby-plugin-sharp: ^4.8.1 => 4.8.1 
    gatsby-source-drupal: ^5.8.1 => 5.8.1 
    gatsby-source-drupal-menu-links: ^3.0.1 => 3.0.1 
    gatsby-transformer-sharp: ^4.8.0 => 4.8.0 
  npmGlobalPackages:
    gatsby-cli: 4.8.0
```

**Comment by Khaledgarbaya** [comment]
Hi @jmolivas ,

Thank for reporting the issue

if you want to have links with `/blog/pathname` you can move the `{nodeArticle.path__alias}.js` file inside a blog folder so your new structure will look like this

```
src/pages
├── blog
  ├──{nodeArticle.path__alias}.js
  └── {nodePage.path__alias}.js
```
and change the slug inside your CMS to just the path of the article. e.g `/my-article` instead of `/blog/my-article`.
As per the doc Gatsby will try to convert the field value to a human readable URL without the special characters so probably there's bug there in the slugify function

Hopefully this will unblock you for and we're gonna look at it in the near future

**Comment by jmolivas** [comment]
Hi @Khaledgarbaya 

Based on your explanation it's not a bug it's a feature ;) 

What you mentioned make sense from the point of view of Gatsby but not sure if that make sense form the point of view of the CMS.

There is not an error on the `path_alias` aka `slug` generation the CMS is generating as `/blog/my-article` and this full path_alias name is a pretty common pattern when generating those using a CMS

Updating the slug at the CMS is doable, but not sure if this make sense with big sites migrating from a CMS (Drupal/WP) to a Decoupled FE tool as Gatsby.

Also CMS with highly complex routes probably won't be able to take full advantage of this feature. I can. probably provide more advanced example but at this very moment I can think of:
> Routes containing data from fields as tags, dates, etc, generating different route patterns from the same Content-Type.

**Comment by raae** [comment]
This is still an issue, and I agree with @jmolivas that it's a bug and not a feature 🤪

**Comment by raae** [comment]
Is this not considered a bug @Khaledgarbaya?
